src/AdoNet/Orleans.Persistence.AdoNet/Storage/Provider/JenkinsHash.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Text;
+
+namespace Orleans.Storage
+{
+    // Based on the version in http://home.comcast.net/~bretm/hash/7.html, which is based on that
+    // in http://burtleburtle.net/bob/hash/evahash.html.
+    // Note that we only use the version that takes three ulongs, which was written by the Orleans team.
+    // implementation restored from Orleans v3.7.2: https://github.com/dotnet/orleans/blob/b24e446abfd883f0e4ed614f5267eaa3331548dc/src/Orleans.Core.Abstractions/IDs/JenkinsHash.cs
+    internal static class JenkinsHash
+    {
+        private static void Mix(ref uint a, ref uint b, ref uint c)
+        {
+            a -= b; a -= c; a ^= (c >> 13);
+            b -= c; b -= a; b ^= (a << 8);
+            c -= a; c -= b; c ^= (b >> 13);
+            a -= b; a -= c; a ^= (c >> 12);
+            b -= c; b -= a; b ^= (a << 16);
+            c -= a; c -= b; c ^= (b >> 5);
+            a -= b; a -= c; a ^= (c >> 3);
+            b -= c; b -= a; b ^= (a << 10);
+            c -= a; c -= b; c ^= (b >> 15);
+        }
+
+        // This is the reference implementation of the Jenkins hash.
+        public static uint ComputeHash(byte[] data)
+        {
+            return ComputeHash(data.AsSpan());
+        }
+
+        // This is the reference implementation of the Jenkins hash.
+        public static uint ComputeHash(ReadOnlySpan<byte> data)
+        {
+            int len = data.Length;
+            uint a = 0x9e3779b9;
+            uint b = a;
+            uint c = 0;
+            int i = 0;
+
+            while (i + 12 <= len)
```

Comment:
`i` gets incremented in the loop. thus the loop-condition nees to be re-evaluated each iteration. Change it to
```suggestion
            while (i <= len - 12)
```
then the loop condition is only evaluated once (and can potentially be kept in a register).

Reply:
done